File: cuckoo-hibersap-example-datamodel/src/main/java/de/akquinet/jbosscc/cuckoo/example/model/CustomerSearch.java
package de.akquinet.jbosscc.cuckoo.example.model;

import org.hibersap.annotations.Bapi;
import org.hibersap.annotations.Import;
import org.hibersap.annotations.Parameter;
import org.hibersap.annotations.Table;

import java.util.List;

@Bapi( "BAPI_FLCUST_GETLIST" )
public class CustomerSearch
{
    @Import
    @Parameter( "CUSTOMER_NAME" )
    private String nameSearchPattern;

    @Import
    @Parameter( "MAX_ROWS" )
    private int maxRows;

    @Table
    @Parameter( "RETURN" )
    private List<ReturnMessage> returnMessages;

    @Table
    @Parameter( "CUSTOMER_LIST" )
    private List<CustomerWithId> customers;

    public CustomerSearch( String nameSearchPattern, int maxRows )
    {
        this.maxRows = maxRows;
        this.nameSearchPattern = nameSearchPattern;
    }

    public List<CustomerWithId> getCustomers()
    {
        return customers;
    }

    public List<ReturnMessage> getReturnMessages()
    {
        return returnMessages;
    }
}

File: src/main/kotlin/org/hibersap/customer/model/CustomerSearch.kt
package org.hibersap.customer.model

import org.hibersap.annotations.Bapi
import org.hibersap.annotations.Import
import org.hibersap.annotations.Parameter
import org.hibersap.annotations.Table
import org.hibersap.annotations.ThrowExceptionOnError

@Bapi("BAPI_FLCUST_GETLIST")
@ThrowExceptionOnError(returnStructure = "TABLE/RETURN")
data class CustomerSearch(

        @Parameter("CUSTOMER_NAME")
        @Import
        private var nameSearchPattern: String,

        @Parameter("MAX_ROWS")
        @Import
        private var maxRows: Int,

        @Table
        @Parameter("RETURN")
        var returnMessages: List<ReturnMessage> = listOf(),

        @Table
        @Parameter("CUSTOMER_LIST")
        var customers: List<CustomerWithId> = listOf()
)